spyder/plugins/outlineexplorer/api.py

```diff
@@ -48,6 +49,23 @@ def document_cells(block, forward=True):
     else:
         block = block.previous()
 
+    if not block.isValid():
+        return
+
+    if cell_list is not None:
```

Comment:
Please add a comment or update the docstring for what cell_list does. I guess it is the caching you mentioned?

Reply:
Done. Yes it is

Reply:
Thanks!